plugins/datastreamer_plugin/utils/dynamic_types_utils.cpp

```diff
@@ -34,149 +35,260 @@ namespace eprosima {
+            actual_kind);    // use last kind (direct parent = parent_data)
+    }
+}
+
+DynamicData_ptr get_parent_data_of_member(
+        const DynamicData_ptr& data,
+        const std::vector<MemberId>& members_tree,
+        const std::vector<TypeKind>& kind_tree,
+        unsigned int array_indexes /* = 0 */)
+{
+    // Get the member and kind of the current level
+    const MemberId& member_id = members_tree[array_indexes];
+    const TypeKind& kind = kind_tree[array_indexes];
+
+    if (array_indexes == members_tree.size() - 1)
+    {
+        // One-to-last value, so this one is the value to take, as it is the parent of the data
+        return data;
+    }
+    else
+    {
+        switch (kind)
+        {
+            // TODO
+        default:
+            // TODO add exception
+            WARNING("Error getting data");
+            return data;
+            break;
+        }
     }
 }
 
 double get_numeric_type_from_data(
-        eprosima::fastrtps::types::DynamicData_ptr data,
-        eprosima::fastrtps::types::MemberId member,
-        eprosima::fastrtps::types::TypeKind kind)
+        const DynamicData_ptr& data,
+        const MemberId& member,
+        const TypeKind& kind)
 {
+    DEBUG("Getting numeric data of kind " << kind << " in member " << member);
+
     switch (kind)
     {
-        case eprosima::fastrtps::types::TK_BOOLEAN:
+        case TK_BOOLEAN:
             return static_cast<double>(data->get_bool_value(member));
 
-        case eprosima::fastrtps::types::TK_BYTE:
+        case TK_BYTE:
             return static_cast<double>(data->get_byte_value(member));
 
-        case eprosima::fastrtps::types::TK_INT16:
+        case TK_INT16:
             return static_cast<double>(data->get_int16_value(member));
 
-        case eprosima::fastrtps::types::TK_INT32:
+        case TK_INT32:
             return static_cast<double>(data->get_int32_value(member));
 
-        case eprosima::fastrtps::types::TK_INT64:
+        case TK_INT64:
             return static_cast<double>(data->get_int64_value(member));
 
-        case eprosima::fastrtps::types::TK_UINT16:
+        case TK_UINT16:
             return static_cast<double>(data->get_uint16_value(member));
 
-        case eprosima::fastrtps::types::TK_UINT32:
+        case TK_UINT32:
             return static_cast<double>(data->get_uint32_value(member));
 
-        case eprosima::fastrtps::types::TK_UINT64:
+        case TK_UINT64:
             return static_cast<double>(data->get_uint64_value(member));
 
-        case eprosima::fastrtps::types::TK_FLOAT32:
+        case TK_FLOAT32:
             return static_cast<double>(data->get_float32_value(member));
 
-        case eprosima::fastrtps::types::TK_FLOAT64:
+        case TK_FLOAT64:
             return static_cast<double>(data->get_float64_value(member));
 
-        case eprosima::fastrtps::types::TK_FLOAT128:
+        case TK_FLOAT128:
             return static_cast<double>(data->get_float128_value(member));
```

Comment:
NIT: I'd prefer a middle ground, like `fastrtps::types::TK_...` to express the underlying enum while not being too verbose